crates/vm/backends/levm/mod.rs

```diff
@@ -300,37 +301,53 @@ impl LEVM {
     pub(crate) fn read_withdrawal_requests(
         block_header: &BlockHeader,
         db: &mut GeneralizedDatabase,
-    ) -> Option<ExecutionReport> {
+    ) -> Result<ExecutionReport, EvmError> {
         let report = generic_system_contract_levm(
             block_header,
             Bytes::new(),
             db,
             *WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS,
             *SYSTEM_ADDRESS,
-        )
-        .ok()?;
+        )?;
+
+        // According to EIP-7002 we need to check if the WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS
+        // has any code after being deployed. If not, the whole block becomes invalid.
+        // https://github.com/ethereum/EIPs/blob/master/EIPS/eip-7002.md
+        let account = db.get_account(*WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS)?;
+        if account.info.code_hash == EMPTY_CODE_HASH {
+            return Err(EvmError::Custom("BlockException.SYSTEM_CONTRACT_EMPTY: WITHDRAWAL_REQUEST_PREDEPLOY_ADDRESS has no code after deployment".to_string()));
+        }
 
         match report.result {
-            TxResult::Success => Some(report),
-            _ => None,
+            TxResult::Success => Ok(report),
+            // EIP-7002 specifies that a failed system call invalidates the entire block.
+            TxResult::Revert(vm_error) => Err(EvmError::from(vm_error)),
```

Comment:
```suggestion
            TxResult::Revert(vm_error) => Err(EvmError::Custom(format!(
                "Read withdrawals request shouldn't revert. VM Error: {}",
                vm_error
            ))),
```
This is just a suggestion of a possible change in the return of the error. The reason behind it is that I think that it may not be enough information just to return the `vm_error` without any explanation around it.
I would just add a little bit of context saying what actually happened, that a system call shouldn't revert or something like that.
Let me know what you think and feel free to change it for what you think is best or keep it as it is if you think is more correct

Reply:
True, the error should give more context. I changed it here [`2025181`](https://github.com/lambdaclass/ethrex/pull/2630/commits/20251818070e7cb2947df335e6a9f1b5a24b2df8). Let me know what you think about it!

Reply:
It's perfect